src/Control/Lens/Cons.hs

```diff
@@ -319,6 +323,42 @@ _tail :: Cons s s a a => Traversal' s s
 _tail = _Cons._2
 {-# INLINE _tail #-}
 
+-- | Convert one type with a 'Cons' instance into the other.
+--
+-- Rewrite rules are provided for efficient conversion between
+-- 'String' and 'Text', @['Word8']@ and 'ByteString', and lazy and
+-- strict 'Text' and 'ByteString'.  Programs must be compiled
+-- with @-O@ to use them.
+--
+-- If @(recons . recons)@ exists it should obey:
+--
+-- > recons . recons ≡ id
+--
+recons :: (Cons s1 s1 a a, Cons s2 s2 a a, AsEmpty s2) => Getter s1 s2
+recons = to (unfoldr uncons)
+{-# NOINLINE [2] recons #-}
+
+{-# RULES
+"recons/id" recons = id
+"recons/string-text" recons = to StrictT.pack
+"recons/text-string" recons = to StrictT.unpack
+"recons/string-lazytext" recons = to LazyT.pack
+"recons/lazytext-string" recons = to LazyT.unpack
+"recons/text-strict" recons = to LazyT.toStrict
+"recons/text-lazy" recons = to LazyT.fromStrict
+"recons/list-bs" recons = to StrictB.pack
+"recons/bs-list" recons = to StrictB.unpack
+"recons/list-lazybs" recons = to LazyB.pack
+"recons/lazybs-list" recons = to LazyB.unpack
+"recons/bs-strict" recons = to LazyB.toStrict
+"recons/bs-lazy" recons = to LazyB.fromStrict
+ #-}
+
+-- | > unfoldr f = foldrOf (unfolded f) cons (_Empty # ())
+--
+unfoldr :: (Cons s2 s2 a a, AsEmpty s2) => (s1 -> Maybe (a, s1)) -> s1 -> s2
```

Comment:
I discussed this change with @ekmett who seemed more keen on `unfoldr` than `recons` :)